src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -4810,7 +4810,7 @@ internal bool EmitNullablePublicOnly
             {
                 if (!_lazyEmitNullablePublicOnly.HasValue())
                 {
-                    bool value = SyntaxTrees.FirstOrDefault()?.Options?.Features?.ContainsKey("nullablePublicOnly") == true;
+                    bool value = SyntaxTrees.FirstOrDefault()?.Options?.HasFeature(CodeAnalysis.Feature.NullablePublicOnly) == true;
```

Comment:
This doesn't seem strictly equivalent to the previous code - `null` value in the feature dictionary would previously be accepted.

There are other changes like this in the PR (replacing `Features?.ContainsKey` with `HasFeature`).

But I guess the new behavior is better, just wanted to point out this is a small breaking change.